pool/paymentmgr.go

```diff
@@ -904,53 +916,64 @@ func (pm *PaymentMgr) payDividends(ctx context.Context, height uint32) error {
 		outs[addr] = amt
 	}
 
-	tx, err := txCreator.CreateRawTransaction(ctx, inputs, outs, nil, nil)
-	if err != nil {
-		return fmt.Errorf("unable to create raw transaction: %v", err)
+	// Ensure the wallet is aware of all the outputs to be spent by the payout
+	// transaction.
+	if maxSpendableHeight < height {
+		maxSpendableHeight = height
 	}
 
-	txB, err := tx.Bytes()
+	hashes := make([]*chainhash.Hash, 0, len(txHashes))
+	for _, hash := range txHashes {
+		hashes = append(hashes, hash)
+	}
+
+	notifSource, err := pm.cfg.GetTxConfNotifications(hashes,
```

Comment:
This seems to be creating a new notification stream every time payDividends is called. Are the old streams properly closed?

Reply:
No it's not, each requested stream terminates at the provided `StopAfter` height. `GetTxNotifications` is only returning the function to read the stream with after the confirmation notification request is sent. This function is the `Recv` of the confirmation notification client.